From a5295384863a6825e72d1443b631072c960e7496 Mon Sep 17 00:00:00 2001
From: Adam Wulw <adam.wulw@intel.com>
Date: Tue, 30 Oct 2018 07:22:17 +0100
Subject: [PATCH 1474/1505] ASoC: core: Do not return for dummy codec for
 BXT_ULL machine

BXT_ULL machine driver is based on dummy codec. DAI link creation
requires connection between cpu and codec DAIs (widgets).
Thus, we need dummy codec widgets (Playback/Capture) to be created.

Change-Id: I8519f5d36b3794265df28f54fecd5ed64f0f8f88
Tracked-On: OAM-88436
Signed-off-by: Mateusz Kalinowski <mateuszx.kalinowski@intel.com>
---
 sound/soc/soc-core.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index 18a7c5e686cd..a5a5f633d277 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -1243,7 +1243,8 @@ static int soc_probe_component(struct snd_soc_card *card,
 	struct snd_soc_dai *dai;
 	int ret;
 
-	if (!strcmp(component->name, "snd-soc-dummy"))
+	if (!strcmp(component->name, "snd-soc-dummy") &&
+		!IS_ENABLED(CONFIG_SND_SOC_INTEL_BXT_ULL_MACH))
 		return 0;
 
 	if (component->card) {
-- 
2.17.1

